include/rsl/static_vector.hpp

```diff
@@ -0,0 +1,124 @@
+#pragma once
+
+#include <rclcpp/parameter_value.hpp>
+#include <tcb_span/span.hpp>
+
+#include <array>
+#include <cassert>
+#include <type_traits>
+#include <vector>
+
+namespace rsl {
+
+/**
+ * @brief A Fixed size array with conversion operators for tcb::span and rclcpp::Parameter.
+ * At compile time you specify the maximum size of the container with the Size template parameter.
+ * At runtime you can populate up to Size values.
+ */
+template <typename ValueType, size_t Size>
+class StaticVector {
+   public:
+    /**
+     * @brief Allow default construction
+     */
+    StaticVector() = default;
+
+    /**
+     * @brief Construct from a Collection type like std::vector
+     */
+    template <typename Collection>
+    StaticVector(Collection const& collection) {
+        assert(collection.size() <= Size);
+        len_ = std::min(collection.size(), Size);
+        std::copy(
+            std::cbegin(collection),
+            std::cbegin(collection) +
+                static_cast<
+                    typename std::remove_reference_t<decltype(collection)>::difference_type>(len_),
```

Comment:
What motivated you to add all this static casting and type trait code? Is adding the `size_t` `len_` not correct or error prone?

Reply:
Technically the difference_type inside a vector is a singed value, and Wconversion complains about adding a size_t to an iterator.